Devdocs: fix Timezone styles in PostSchedule example

tofumatt opened the pull request:
Fixes #22966

Note there were a few hard-code `style` props here as well messing up the padding. I removed those too, but I didn't make a separate issue for it as I didn't notice until mid-way through the patch 😄 

### Before
<img width="735" alt="screenshot 2018-03-03 10 18 56" src="https://user-images.githubusercontent.com/90871/36938691-31bff2aa-1f1d-11e8-99b7-b27fee82ef41.png">

### After
<img width="623" alt="screenshot 2018-03-03 19 05 11" src="https://user-images.githubusercontent.com/90871/36938689-2d9a16ec-1f1d-11e8-9bdf-4f63a3c062d1.png">

andfinally commented:
I'm still earning my stripes, so I better not approve, but this looks good to me, and renders OK - I gave it a try on IE11 Win 7 and Chrome Android v7 as well as Chrome macOS. I find the alignment of the stacked items a bit strange on the Android version, but that's unrelated.

IE11:

![ie11](https://user-images.githubusercontent.com/1647564/36950514-d4c32bf8-1fee-11e8-9f82-6cd77e45d601.png)

Android:

![android-v7](https://user-images.githubusercontent.com/1647564/36950513-cfcaf2ac-1fee-11e8-9a93-2ef1e9a39a02.png)

tofumatt commented:
Hmmm, true, I'll give that a tweak and push a fix later today.

Also looks like I have a lint error I need to fix too, oops.

tofumatt commented:
Updated the styles after noticing the mobile ones are quite weird. I cleaned them up too and put the CSS in a stylesheet rather than as react props, which meant there are now smaller and larger viewport styles and these actually grow with screen size a bit better. Here are the new styles:

### Mobile
![screen shot 2018-03-05 at 22 06 38](https://user-images.githubusercontent.com/90871/37002989-5e72eee0-20c3-11e8-909d-9f15b287492a.png)

### Desktop
<img width="740" alt="screenshot 2018-03-05 22 01 11" src="https://user-images.githubusercontent.com/90871/37002998-6956d1dc-20c3-11e8-82c1-7f9df9711dc9.png">

tofumatt commented:
Just flagging the people GitHub thinks would be good to review this, but I'm thinking it only needs one review; it's pretty basic 😄

tyxla commented on client/devdocs/style.scss:
```diff
@@ -199,3 +199,28 @@
+.docs__post-schedule {
+	@include breakpoint( ">480px" ) {
+		display: flex;
+		justify-content: space-between;
```
Is this necessary for anything?

tyxla commented on client/devdocs/style.scss:
```diff
@@ -199,3 +199,28 @@
+}
+
+.docs__post-schedule-card {
+	display: inline-block;
```
With the `display: flex` the `display: inline-block` shouldn't be necessary anymore, I think?

tyxla commented on client/devdocs/style.scss:
```diff
@@ -199,3 +199,28 @@
+.docs__post-schedule-card {
+	display: inline-block;
+	margin: 0;
+	vertical-align: top;
```
I'm not sure if this is necessary.

tyxla commented on client/devdocs/style.scss:
```diff
@@ -199,3 +199,28 @@
+.docs__post-schedule-timezone-card-wrapper {
+	display: inline-block;
+	margin: 15px 0 0;
+	vertical-align: top;
```
I'm not sure if this is necessary.

tyxla commented on client/devdocs/style.scss:
```diff
@@ -199,3 +199,28 @@
+}
+
+.docs__post-schedule-timezone-card-wrapper {
+	display: inline-block;
```
With the `display: flex` the `display: inline-block` shouldn't be necessary anymore, I think?

tyxla commented on client/components/timezone/index.jsx:
```diff
@@ -60,16 +61,24 @@ class Timezone extends Component {
+			// warning.
+			/* eslint-disable jsx-a11y/no-onchange */
+			<select
+				className={ classnames( 'timezone__component', className ) }
```
I've never seen using `__component` in a selector. Our [CSS coding guidelines](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/css.md) suggest that the part after the double underscores define the element, so I'd rather use `__main` instead. In some cases it makes sense to use `__wrapper`, but this is not really a wrapper, so maybe `__main` makes more sense. 

What do you think?

tofumatt replied:
I'm still getting to know the selector naming rules–they're a bit different from what I'm used to 😄 

I probably just messed up; I'll have another look at the rules and try to come up with a better name.

tyxla commented on client/components/timezone/index.jsx:
```diff
@@ -60,16 +61,24 @@ class Timezone extends Component {
+		const { className, selectedZone } = this.props;
 		return (
-			<select onChange={ this.onSelect } value={ selectedZone || '' }>
+			// Changing this to onBlur will not work, so we ignore this
```
I think that regardless of the reasons behind not being able to migrate `onChange` to `onBlur`, we shouldn't encourage this and should be looking into fixing it in the future. Personally I'd pay the price and accept single failing tests (because of eslines) in favor of not adding an `eslint-disable` rule when it indicates a real a11y flaw.

In case you decide to go with the rule ignore, could you elaborate in the inline comment and explain why it won't work? Possibly add a `TODO` too? It could make a good case to revisit in the future.

tyxla commented on client/components/post-schedule/style.scss:
```diff
@@ -138,6 +138,10 @@ input[type="text"].post-schedule__clock-time {
 }
 
+.post-schedule__component-timezone {
+	width: 100%;
```
This is only used in the example; does it make sense to be in the component styles?

tyxla commented on client/devdocs/style.scss:
```diff
@@ -199,3 +199,28 @@
 	}
 }
+
+.docs__post-schedule {
```
I think the main devdocs stylesheet is not the right place to introduce component-specific styles.

tofumatt commented:
@tyxla I totally agree re: keeping styles out of the main stylesheet. I put it there as there isn't a component-specific stylesheet paradigm but I agree and want that too 😄. I figured introducing that in this patch was too much, but I wanted some media queries and putting them inline in the component seemed worse 😓

Fair enough though, I'll try to focus this PR a bit more and we'll see about the doc-specific styles in a future patch. 👍

tofumatt commented:
I think realistically not. I wasn't sure where to introduce component-specific styles. It still doesn't have merge conflicts though so I could see about tweaking it soon.
